From 9314330be83326e482e88d54f1d4ecd52843b5a6 Mon Sep 17 00:00:00 2001
From: vijayabharathi shetty <vijayabharathix.shetty@intel.com>
Date: Wed, 21 Jun 2023 13:20:03 +0000
Subject: [PATCH] CTL: Remove unsupported power and energy sensors

For catlow, excluded polling of unsupported Energy and cpu related
power sensors. The unsupported and excluded sensors are,
1) Cpu Energy CPU1
2) Dimm Energy CPU1 (PECI index 4 is not supported)
3) Cpu Power Average
4) Cpu Power Cap CP

PECI index 3 gives the DIMM Power Average value. So added that change.

Before changes:
Cpu Energy CPU1  | 1344.000 | unspecified |ok|na|na| na | na | na | na
Dimm Energy CPU1 | na       | unspecified |na|na|na| na | na | na | na
CPU Power        | 0.000    | Watts | ok | na | na | na | na | na | na
Cpu Power Averag | 4.000    | Watts | ok | na | na | na | na | na | na
Cpu Power Cap CP | 0.000    | Watts | ok | na | na | na | na | na | na
Dimm Power Avera | na       | Watts | ok | na | na | na | na | na | na
Dimm Power Cap C | 0.000    | Watts | ok | na | na | na | na | na | na
Memory Power     | 0.000    | Watts | ok | na | na | na | na | na | na
Total Power      | 32.000   | Watts | ok | na | na | na | na | na | na

After changes:
CPU Power        | 0.000  | Watts | ok | na | na | na | na | na | na
Dimm Power Avera | 4.000  | Watts | ok | na | na | na | na | na | na
Dimm Power Cap C | 0.000  | Watts | ok | na | na | na | na | na | na
Memory Power     | 0.000  | Watts | ok | na | na | na | na | na | na
Total Power      | 32.000 | Watts | ok | na | na | na | na | na | na

Signed-off-by: vijayabharathi shetty <vijayabharathix.shetty@intel.com>
Upstream-Status: Pending
---
 Makefile                       |  4 ++++
 drivers/hwmon/Makefile         |  2 ++
 drivers/hwmon/peci-dimmpower.c | 10 ++++++++++
 3 files changed, 16 insertions(+)

diff --git a/Makefile b/Makefile
index fcee25420bf9..df1f8e0825ab 100644
--- a/Makefile
+++ b/Makefile
@@ -932,6 +932,10 @@ KBUILD_CFLAGS	+= $(CC_FLAGS_FTRACE) $(CC_FLAGS_USING)
 KBUILD_AFLAGS	+= $(CC_FLAGS_USING)
 endif
 
+ifneq ($(CTL),0)
+KBUILD_CFLAGS += -DCTL_FEATURE_DISABLE
+endif
+
 # We trigger additional mismatches with less inlining
 ifdef CONFIG_DEBUG_SECTION_MISMATCH
 KBUILD_CFLAGS += -fno-inline-functions-called-once
diff --git a/drivers/hwmon/Makefile b/drivers/hwmon/Makefile
index 07fae2bbbe8e..54192f927b6b 100644
--- a/drivers/hwmon/Makefile
+++ b/drivers/hwmon/Makefile
@@ -166,7 +166,9 @@ obj-$(CONFIG_SENSORS_PC87427)	+= pc87427.o
 obj-$(CONFIG_SENSORS_PCF8591)	+= pcf8591.o
 obj-$(CONFIG_SENSORS_PECI_CPUTEMP_LEGACY)	+= peci-cputemp.o
 obj-$(CONFIG_SENSORS_PECI_DIMMTEMP_LEGACY)	+= peci-dimmtemp.o
+ifeq ($(CTL),0)
 obj-$(CONFIG_SENSORS_PECI_CPUPOWER_LEGACY)	+= peci-cpupower.o
+endif
 obj-$(CONFIG_SENSORS_PECI_DIMMPOWER_LEGACY)	+= peci-dimmpower.o
 obj-$(CONFIG_SENSORS_PECI_PLATFORMPOWER_LEGACY) += peci-platformpower.o
 obj-$(CONFIG_SENSORS_POWR1220)  += powr1220.o
diff --git a/drivers/hwmon/peci-dimmpower.c b/drivers/hwmon/peci-dimmpower.c
index 9500b5dd5e2d..33349cc677de 100644
--- a/drivers/hwmon/peci-dimmpower.c
+++ b/drivers/hwmon/peci-dimmpower.c
@@ -95,9 +95,15 @@ peci_dimmpower_get_energy_counter(struct peci_dimmpower *priv,
 		goto unlock;
 	}
 
+#ifndef CTL_FEATURE_DISABLE
 	ret = peci_pcs_read(priv->mgr, PECI_MBX_INDEX_ENERGY_STATUS,
 			    PECI_PKG_ID_DIMM, (u8 *)&sensor_data->uvalue,
 			    sizeof(u32));
+#else
+	        ret = peci_pcs_read(priv->mgr, PECI_MBX_INDEX_ENERGY_COUNTER,
+                            PECI_PKG_ID_DIMM, (u8 *)&sensor_data->uvalue,
+                            sizeof(u32));
+#endif
 	if (ret) {
 		dev_dbg(priv->dev, "not able to read dimm energy\n");
 		goto unlock;
@@ -321,6 +327,7 @@ peci_dimmpower_read_min_power(void *ctx, struct peci_sensor_conf *sensor_conf,
 	return 0;
 }
 
+#ifndef CTL_FEATURE_DISABLE
 static int
 peci_dimmpower_read_energy(void *ctx, struct peci_sensor_conf *sensor_conf,
 			   struct peci_sensor_data *sensor_data)
@@ -371,6 +378,7 @@ peci_dimmpower_read_energy(void *ctx, struct peci_sensor_conf *sensor_conf,
 	mutex_unlock(&sensor_data->lock);
 	return ret;
 }
+#endif
 
 static struct peci_sensor_conf
 peci_dimmpower_power_cfg[PECI_DIMMPOWER_POWER_CHANNEL_COUNT]
@@ -413,6 +421,7 @@ peci_dimmpower_energy_cfg[PECI_DIMMPOWER_ENERGY_CHANNEL_COUNT]
 			 [PECI_DIMMPOWER_ENERGY_SENSOR_COUNT] = {
 	/* Channel 0  - Energy */
 	{
+#ifndef CTL_FEATURE_DISABLE
 		{
 			.attribute = hwmon_energy_input,
 			.config = HWMON_E_INPUT,
@@ -420,6 +429,7 @@ peci_dimmpower_energy_cfg[PECI_DIMMPOWER_ENERGY_CHANNEL_COUNT]
 			.read = peci_dimmpower_read_energy,
 			.write = NULL,
 		},
+#endif
 	}
 };
 
-- 
2.17.1

